### PR TITLE
Update to Bevy 0.12

### LINKED ISSUE #59: Prepare for Bevy 0.12

### PR DESCRIPTION
**Pull request by GitGhillie**
Closes #59 
The only relevant migration seems to be:
FixedTime to Time (Bevy pull 8964)

To do:

- [x] Test Doppler
- [x] Update Bevy dependency to 0.12
- [x] Remove smooth-bevy-cameras dev dependency so we don't have to wait for it (it got updated)

### CODE REVIEW

**Comment by GitGhillie** [comment]
Blocked on Bevy 0.12 release and #61, other than that it should be good to go

**Comment by GitGhillie** [comment]
Not blocking on 61 anymore since smooth-bevy-cameras got updated to 0.12

**Comment by pop** [comment]
@GitGhillie I think the [README](https://github.com/Salzian/bevy_fmod#versioning) also needs to be updated if this version is compatible with Bevy 0.12.

**Comment by GitGhillie** [comment]
Thanks, I forgot about that! @Salzian did you want to do a final release before switching to 0.12? That will affect the version table.

File: src/components/velocity.rs [hunk 1]
```diff
@@ -16,21 +16,18 @@ pub struct Velocity {
...
-            velocity.current_velocity = delta_position / time.period.as_secs_f32();
+            velocity.current_velocity = delta_position / time.delta().as_secs_f32();
             velocity.last_position = current_position;
         })
     }
 }
 
 impl Plugin for VelocityPlugin {
     fn build(&self, app: &mut App) {
-        app.add_systems(Update, Self::update_velocity);
```
Comment (Salzian) [inline src/components/velocity.rs:34]: Why this change? Doesn't this reintroduce the glitchy doppler effect?

https://github.com/Salzian/bevy_fmod/commit/8af8e8f666dcb006bc53260179359641ecf9ee7b#diff-819c727ab5d1508bb4a39ccd5fad25915fe88923d71f1d3c13dccb6131f839daL35

Comment (GitGhillie) [reply src/components/velocity.rs:34]: Hmm I thought it didn't but I just checked again with a sine wave and it does :( guess it's more subtle than I remembered.

Anyways, I think we were mis-using the time API previously, so with the 0.12 migration I couldn't figure out how to replicate the same behavior. Because I think we were actually using a fixed time in the Update schedule, which does not have a fixed time.

I'll need more time to look into this

Comment (GitGhillie) [reply src/components/velocity.rs]: Tried something, feels like a clunky solution but it's way better. Not a 100% as good as old but you can barely hear it with a pure sine wave

Comment (GitGhillie) [reply src/components/velocity.rs]: What it sounds like https://youtu.be/ckAXzCvEJAY?feature=shared

**Comment by pop** [comment]
@GitGhillie @Salzian I can use this branch for now, but I would really appreciate this PR being merged so it officially supports Bevy 0.12. Is there anything I can do to help? Testing, etc?

File: src/components/velocity.rs (hunk 1 above)
Comment (GitGhillie) [reply src/components/velocity.rs]: > What it sounds like https://youtu.be/ckAXzCvEJAY?feature=shared

I also tried this example with bevy_framepace and it made it a lot worse. This is a hint in what is going wrong but I haven't had the time yet to find out exactly what it is

**Comment by GitGhillie** [comment]
Delaying the delta time by one frame fixed the unstable Doppler effect

**Comment by Salzian** [comment]
Accepting this for now, as I can't reproduce the unstable Doppler effect.

**Review by Salzian — APPROVED** [review]
